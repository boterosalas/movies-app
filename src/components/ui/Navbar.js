import React from 'react'
import { Link, NavLink } from 'react-router-dom'

const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <div className="container-fluid">
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav w-100 justify-content-between">
                        <li>
                            <ul className="navbar-nav">
                                <li className="nav-item">
                                    <NavLink
                                        activeClassName="active"
                                        className="nav-item nav-link"
                                        exact
                                        to="/characters"
                                    >
                                        Characters
                                    </NavLink>
                                </li>
                                <li className="nav-item">
                                    <NavLink
                                        activeClassName="active"
                                        className="nav-item nav-link"
                                        exact
                                        to="/comics"
                                    >
                                        Comics
                                    </NavLink>
                                </li>
                                <li className="nav-item">
                                    <NavLink
                                        activeClassName="active"
                                        className="nav-item nav-link"
                                        exact
                                        to="/series"
                                    >
                                        Series
                                    </NavLink>
                                </li>
                                <li className="nav-item">
                                    <NavLink
                                        activeClassName="active"
                                        className="nav-item nav-link"
                                        exact
                                        to="/events"
                                    >
                                        Events
                                    </NavLink>
                                </li>
                                <li className="nav-item">
                                    <NavLink
                                        activeClassName="active"
                                        className="nav-item nav-link"
                                        exact
                                        to="/stories"
                                    >
                                        Stories
                                    </NavLink>
                                </li>
                            </ul>
                        </li>

                        <li className="nav-item">
                            <NavLink
                                activeClassName="active"
                                className="nav-item nav-link"
                                exact
                                to="/login"
                            >
                                Logout
                            </NavLink>
                        </li>
                    </ul>
                </div>
            </div>
        </nav>
    )
}

export default Navbar;