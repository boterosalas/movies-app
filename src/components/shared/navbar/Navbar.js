import React, { useContext, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { useHistory } from "react-router-dom";
import { UserContext } from '../../../routers/UserContext';

const Navbar = () => {
    let history = useHistory();
    const { setUser } = useContext(UserContext);
    const [textSearch, setTextSearch] = useState('');

    const searchMovie = (e) => {
        e.preventDefault();
        if (textSearch.trim().length === 0) return;
        history.push(`/search/${textSearch}`);
    }

    const onLogoutClick = () => {
        setUser(null);
        history.push('/login');
    }

    return (
        <nav className="navbar fixed-top navbar-expand-sm navbar-dark bg-dark mb-5">
            <NavLink
                activeClassName="active"
                className="navbar-brand"
                exact
                to="/"
            >
                MoviesApp
            </NavLink>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
            </button>

            <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav mr-auto"></ul>
                <form className="form-inline my-2 my-lg-0" onSubmit={searchMovie}>
                    <input value={textSearch} onChange={(e) => { setTextSearch(e.target.value) }} className="form-control mr-sm-2" type="text" placeholder="Search a movie" aria-label="Search" />
                    <button className="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
                </form>
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <span
                            className="nav-item nav-link pointer"
                            onClick={onLogoutClick}
                        >
                            Logout
                        </span>
                    </li>
                </ul>
            </div>
        </nav>
    )
}

export default Navbar;