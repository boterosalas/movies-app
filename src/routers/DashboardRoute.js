import React, { useContext, useEffect } from 'react';
import { Redirect, Route, Switch, useHistory } from 'react-router-dom';
import Footer from '../components/shared/footer/Footer';
import HomeScreen from '../components/home/HomeScreen';
import MovieScreen from '../components/movie/MovieScreen';
import SearchScreen from '../components/search/SearchScreen';
import Navbar from '../components/shared/navbar/Navbar';
import { UserContext } from './UserContext';
import './DashboardRoute.css';

const DashboardRoute = () => {
    let history = useHistory();
    const { user } = useContext(UserContext);
    useEffect(() => {
        if (!user) {
            history.push('/login');
        }
    }, [user, history])
    return (
        <>
            {
                user &&
                <>
                    <Navbar />
                    <main className="container mt-7">
                        <Switch>
                            <Route exact path="/" component={HomeScreen} />
                            <Route exact path="/search/:text" component={SearchScreen} />
                            <Route exact path="/movie/:id" component={MovieScreen} />
                            <Redirect to="/" />
                        </Switch>
                    </main>
                    <Footer />
                </>
            }
        </>
    )
}

export default DashboardRoute
