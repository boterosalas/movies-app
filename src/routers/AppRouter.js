import React, { useState } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
} from "react-router-dom";
import LoginScreen from '../components/login/LoginScreen';
import DashboardRoute from './DashboardRoute';
import { UserContext } from './UserContext';


const AppRouter = () => {
    const [user, setUser] = useState(null);
    return (
        <UserContext.Provider value={{ user, setUser }} >
            <Router>
                <div>
                    <Switch>
                        <Route exac path="/login" component={LoginScreen} />
                        <Route exac path="/" component={DashboardRoute} />
                    </Switch>
                </div>
            </Router>
        </UserContext.Provider>
    )
}

export default AppRouter
