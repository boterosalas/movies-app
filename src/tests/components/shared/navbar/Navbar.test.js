import React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';
import Navbar from '../../../../components/shared/navbar/Navbar';
import { UserContext } from '../../../../routers/UserContext';
import { MemoryRouter, Router } from 'react-router-dom';


describe('Test on <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    };

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{ setUser }} >
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </UserContext.Provider>
    );

    test('Should display the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should redirect to a movie search when enter key be press', () => {
        const movie = 'avengers';
        wrapper.find('input').simulate('change', { target: { value: movie } });
        wrapper.find('form').simulate('submit');
        expect(historyMock.push).toHaveBeenCalledWith(`/search/${movie}`);
    });

    test('Should logout', () => {
        wrapper.find('#logout-button').simulate('click');
        expect(setUser).toHaveBeenCalledWith(null);
        expect(historyMock.push).toHaveBeenCalledWith('/login');
    });


})