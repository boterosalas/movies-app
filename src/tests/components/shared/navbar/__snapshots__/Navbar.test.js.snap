// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Navbar /> Should display the component correctly 1`] = `
<nav
  className="navbar fixed-top navbar-expand-sm navbar-dark bg-dark mb-5"
>
  <a
    aria-current={null}
    className="navbar-brand"
    onClick={[Function]}
  >
    MoviesApp
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    />
    <form
      className="form-inline my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        onChange={[Function]}
        placeholder="Buscar una pelÃ­cula"
        type="text"
        value=""
      />
      <button
        className="btn btn-outline-success my-2 my-sm-0"
        type="submit"
      >
        Buscar
      </button>
    </form>
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <span
          className="nav-item nav-link pointer"
          id="logout-button"
          onClick={[Function]}
        >
          Salir
        </span>
      </li>
    </ul>
  </div>
</nav>
`;
